modules/nf-core/vg/construct/main.nf

```diff
@@ -0,0 +1,68 @@
+process VG_CONSTRUCT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::vg=1.45.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vg:1.45.0--h9ee0642_0':
+        'quay.io/biocontainers/vg:1.45.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(vcfs), path(tbis), path(msa), path(insertions_fasta)
+    tuple val(meta2), path(fasta)
```

Comment:
```suggestion
    tuple val(meta), path(vcfs), path(tbis)     // Basic inputs for 'vcf mode'
    tuple val(meta2), path(fasta)                    // .. other vcf mode required input
    tuple val(meta3), path(fasta_fai)             //
    tuple val(meta4), path(msa)                     // Basic inputs for 'msa mode' 
    tuple val(meta5), path(insertions_fasta) // Optional inputs for both ethods
```
(sorry, couldn't figure out how to include the fai below the other comment!)

I feel like the inputs for different mutually exclusive modes should be separated, otherwise you'll always be forced to edit channels with `.map{}` etc to insert empty elements relating to the mode you don't want to use. This way you just have to add empty channels (`[]`) which feels simpler to me. 

But it's very subjective of me and I assume you'll be the main user, so I won't fight you on it!

Reply:
I actually wonder if two different modules might be a simpler way of managing this complexity.

Reply:
Although maybe I could change the `vcfs` to `input` and check what file has been given to see what mode it should be run in

Reply:
I'm not sure that's much of an improvement, now there's `tbis` which don't apply to msa and would need an empty element in the channel.

I'm really not sure what the best way is, maybe a maintainer with more experience has a more cunning plan.

Reply:
Yes, there's not really another option I think in this case. I'll ask in Slack for a second opinion, thanks for the review!!